keras_cv/models/backbones/resnet_v2/resnet_v2_backbone.py

```diff
@@ -341,19 +343,22 @@ def __init__(
         if stackwise_dilations is None:
             stackwise_dilations = [1] * num_stacks
 
-        stack_level_outputs = {}
+        pyramid_level_inputs = {}
         for stack_index in range(num_stacks):
+            num_blocks = stackwise_blocks[stack_index]
             x = apply_stack(
                 x,
                 filters=stackwise_filters[stack_index],
-                blocks=stackwise_blocks[stack_index],
+                blocks=num_blocks,
                 stride=stackwise_strides[stack_index],
                 dilations=stackwise_dilations[stack_index],
                 block_type=block_type,
                 first_shortcut=(block_type == "block" or stack_index > 0),
                 stack_index=stack_index,
             )
-            stack_level_outputs[stack_index + 2] = x
+            pyramid_level_inputs[
+                stack_index + 2
+            ] = f"v2_stack_{stack_index}_block{num_blocks}_out"
```

Comment:
(optional, lmk what you think):

It seems a bit fragile to reconstruct this layer name here.
Why don't we just automagically extract it using `x.node.layer.name`?
(As far as I understand, this should always get us the name of the last layer applied to `x`)

Maybe there's a reason this is a bad idea -- @fchollet would know.

Reply:
Wow, just learned something! Had no idea that was possible!!!